content/threads_intro.tex

```diff
@@ -52,8 +52,8 @@
 In addition, a blocked thread will not prevent the progress of \openshmem calls
 performed on other \acp{PE}.
 
-\item In the \CONST{SHMEM\_THREAD\_MULTIPLE} thread level, all \openshmem calls are thread-safe,
-i.e., two concurrently running threads may make \openshmem calls and the outcome
+\item In the \CONST{SHMEM\_THREAD\_MULTIPLE} thread level, all \openshmem calls are thread-safe.
+Any two concurrently running threads may make \openshmem calls and the outcome
 will be as if the calls executed in some order, even if their execution is interleaved.
```

Comment:
Does the "interleaved" semantic imply that if two threads in a PE perform blocking puts to the same buffer at a given target PE, that the result should be the data sent by one of the threads (i.e. not a mixture of the data from both or just garbage)?

Reply:
No, that's still a data race and is not an artifact of threading. Even for a single threaded program, if a PE performs two put operations to the same destination buffer using two different source buffers without an ordering operation in between the puts, then the contents of the destination buffer are not well defined.